lib/std/io_uring.zig

```diff
@@ -526,7 +526,7 @@ pub const IO_Uring = struct {
         const res = linux.io_uring_register(
             self.fd,
             .REGISTER_FILES,
-            fds.ptr,
+            @ptrCast(*const c_void, fds.ptr),
```

Comment:
What could go wrong if we didn't have this @ptrCast()?

Also, the kernel uses `*int` for the fds array on the other side. Should we be using `fds: []const c_int` instead of `fds: []const i32` in the function signature for `register_files()`?

Reply:
We have `fd_t`

Reply:
Yes, and `os.fd_t` maps to `i32`. But should we not be using `c_int` for ABI compatibility since we need to be compatible with the kernel? I guess `c_int` is always going to be the same size but never say never.

Reply:
The type of fd_t should come from os/bits that architecture specific. It must be correct

Reply:
Yes, `os.fd_t` is certainly correct for that use case within Zig in general, according to various architectures, but I am wondering about the signature of a single syscall here, i.e. I would like someone to check how we're passing `[]const os.fd_t` to `io_uring_register()` which on the kernel side expects `[]const c_int`, see: https://github.com/torvalds/linux/blob/v5.8/fs/io_uring.c#L6629

Reply:
The type of the pointer should not bother you since what happens to it anyway
```zig
pub fn io_uring_register(fd: i32, opcode: IORING_REGISTER, arg: ?*const c_void, nr_args: u32) usize {
    return syscall4(.io_uring_register, @bitCast(usize, @as(isize, fd)), @enumToInt(opcode), @ptrToInt(arg), nr_args);
}
```
What matters is memory layout of the array itself, pointer is just a number. The type of arg is a placeholder most likely.

Reply:
> What matters is memory layout of the array itself

Yes, that's what I was wondering. Our array of `int32_t`'s (what `os.fd_t` looks like in memory) is always going to be an array of 4 byte elements, almost always the same as an array of `int`'s, which is what the kernel actually uses in `io_sqe_files_register()`, although an `int` is not precisely the same thing in memory always as an `int32_t` I think, there may be some obscure architectures where an `int` is not always an `int32_t`? Otherwise, we're cool!

Reply:
I'm not aware of any system where int is not 32 bit. To be extra safe we can add `comptime assert(@sizeOf(c_int) == @sizeOf(i32));` where fd_t is defined.

Reply:
Done